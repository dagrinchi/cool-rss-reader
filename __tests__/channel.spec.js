import React from 'react'
import renderer from 'react-test-renderer'

import preloadAll from "jest-next-dynamic";

const TEST_FEED = {"items":[{"title":"These step-back 3s are sent from another planet","link":"http://www.espn.com/nba/story/_/id/26118445/these-step-back-3s-sent-another-planet","pubDate":"Sun, 3 Mar 2019 09:07:27 EST","content":"How is James Harden doing this? Get ready -- there's more coming.","contentSnippet":"How is James Harden doing this? Get ready -- there's more coming.","guid":"26118445","isoDate":"2019-03-03T14:07:27.000Z"},{"title":"Iguodala: Curry still 'underrated,' No. 2 PG ever","link":"http://www.espn.com/nba/story/_/id/26124916/iguodala-curry-underrated-no-2-pg-ever","pubDate":"Sun, 3 Mar 2019 01:05:35 EST","content":"Warriors swingman Andre Iguodala said Saturday night that two-time MVP Stephen Curry still \"doesn't really get the respect he deserves,\" calling his star teammate \"the second-best point guard ever\" behind Magic Johnson.","contentSnippet":"Warriors swingman Andre Iguodala said Saturday night that two-time MVP Stephen Curry still \"doesn't really get the respect he deserves,\" calling his star teammate \"the second-best point guard ever\" behind Magic Johnson.","guid":"26124916","isoDate":"2019-03-03T06:05:35.000Z"},{"title":"Lakers' playoff odds crash after latest stumble","link":"http://www.espn.com/chalk/story/_/id/26127986/los-angeles-lakers-playoff-odds-crash-latest-stumble","pubDate":"Sun, 3 Mar 2019 11:17:25 EST","content":"The Lakers have fallen to their worst postseason odds yet this season, listed at +600 to reach the playoffs following Saturday's loss to the Suns.","contentSnippet":"The Lakers have fallen to their worst postseason odds yet this season, listed at +600 to reach the playoffs following Saturday's loss to the Suns.","guid":"26127986","isoDate":"2019-03-03T16:17:25.000Z"},{"title":"Questionable plays doom 76ers late vs. Warriors","link":"http://www.espn.com/nba/story/_/id/26125031/questionable-plays-doom-76ers-late-vs-warriors","pubDate":"Sun, 3 Mar 2019 01:25:38 EST","content":"Two head-scratching decisions by the 76ers saw them fail to get a potential tying shot off inside the final minute of Saturday night's loss to the Warriors.","contentSnippet":"Two head-scratching decisions by the 76ers saw them fail to get a potential tying shot off inside the final minute of Saturday night's loss to the Warriors.","guid":"26125031","isoDate":"2019-03-03T06:25:38.000Z"},{"title":"Celtics owner: 'Worst February' I can remember","link":"http://www.espn.com/nba/story/_/id/26126958/celtics-owner-worst-february-remember","pubDate":"Sun, 3 Mar 2019 06:56:15 EST","content":"Celtics co-owner Wyc Grousbeck said his team just finished the \"worst February\" he can remember after going 5-6 and finishing on a four-game losing streak.","contentSnippet":"Celtics co-owner Wyc Grousbeck said his team just finished the \"worst February\" he can remember after going 5-6 and finishing on a four-game losing streak.","guid":"26126958","isoDate":"2019-03-03T11:56:15.000Z"},{"title":"Mitchell's career night spurs Jazz's 17-point rally","link":"http://www.espn.com/nba/story/_/id/26125355/mitchell-career-night-spurs-jazz-17-point-rally","pubDate":"Sun, 3 Mar 2019 02:00:24 EST","content":"Jazz star Donovan Mitchell was not sure if a night that included 46 points and a win over the league-leading Bucks amounted to the best game of his career.","contentSnippet":"Jazz star Donovan Mitchell was not sure if a night that included 46 points and a win over the league-leading Bucks amounted to the best game of his career.","guid":"26125355","isoDate":"2019-03-03T07:00:24.000Z"},{"title":"Buss: Reports of Lakers' AD pursuit 'fake news'","link":"http://www.espn.com/nba/story/_/id/26123155/reports-lakers-ad-pursuit-fake-news","pubDate":"Sat, 2 Mar 2019 20:49:19 EST","content":"Lakers owner Jeanie Buss said the media's coverage of the team's pursuit of Anthony Davis was \"irresponsible and frustrating\" and ultimately false.","contentSnippet":"Lakers owner Jeanie Buss said the media's coverage of the team's pursuit of Anthony Davis was \"irresponsible and frustrating\" and ultimately false.","guid":"26123155","isoDate":"2019-03-03T01:49:19.000Z"},{"title":"Who had the best sneakers of Week 20 in the NBA?","link":"http://www.espn.com/nba/story/_/id/26113077/which-player-had-best-sneakers-week-20-nba","pubDate":"Fri, 1 Mar 2019 15:09:20 EST","content":"PJ Tucker paid how much for his kicks?  Also, the return of Isaiah Thomas highlights Week 20 in sneakers.","contentSnippet":"PJ Tucker paid how much for his kicks?  Also, the return of Isaiah Thomas highlights Week 20 in sneakers.","guid":"26113077","isoDate":"2019-03-01T20:09:20.000Z"},{"title":"NBA draft lottery: Zion favorites, traded picks and tank races","link":"http://www.espn.com/nba/story/_/id/26092152/nba-draft-lottery-zion-favorites-traded-picks-tank-races","pubDate":"Thu, 28 Feb 2019 08:54:47 EST","content":"The race to the top of the NBA draft is different this season. Here are our projections.","contentSnippet":"The race to the top of the NBA draft is different this season. Here are our projections.","guid":"26092152","isoDate":"2019-02-28T13:54:47.000Z"},{"title":"Lakers seek answers to loss to NBA-worst Suns","link":"http://www.espn.com/nba/story/_/id/26125829/lakers-seek-answers-loss-nba-worst-suns","pubDate":"Sun, 3 Mar 2019 03:19:39 EST","content":"In a game they considered a must-win, the Lakers hit a new low Saturday night by falling 118-109 to the 13-51 Suns, the league's worst team.","contentSnippet":"In a game they considered a must-win, the Lakers hit a new low Saturday night by falling 118-109 to the 13-51 Suns, the league's worst team.","guid":"26125829","isoDate":"2019-03-03T08:19:39.000Z"},{"title":"Lowe: The Lakers have no excuses","link":"http://www.espn.com/nba/story/_/id/26104826/10-things-like-including-lakers-tank-talk","pubDate":"Thu, 28 Feb 2019 13:35:47 EST","content":"This week we highlight LeBron's challenge, the Anthony Davis situation, Trae Young and a young Knick.","contentSnippet":"This week we highlight LeBron's challenge, the Anthony Davis situation, Trae Young and a young Knick.","guid":"26104826","isoDate":"2019-02-28T18:35:47.000Z"},{"title":"The Warriors might have the most talented starting five ever","link":"http://www.espn.com/nba/story/_/id/26113320/the-warriors-most-talented-starting-five-ever","pubDate":"Fri, 1 Mar 2019 16:09:18 EST","content":"The return of DeMarcus Cousins gave Golden State a lineup whose members could all end up in the Hall of Fame. But the results so far have been mixed.","contentSnippet":"The return of DeMarcus Cousins gave Golden State a lineup whose members could all end up in the Hall of Fame. But the results so far have been mixed.","guid":"26113320","isoDate":"2019-03-01T21:09:18.000Z"},{"title":"Happy 20th, Luka! How the Mavs phenom left his mark as an NBA teen","link":"http://www.espn.com/nba/story/_/id/26099269/happy-20th-luka-how-mavs-phenom-left-mark-nba-teen","pubDate":"Wed, 27 Feb 2019 23:28:28 EST","content":"Luka Doncic spent his first 57 career games putting up numbers not seen since the days of a young LeBron James, Kobe Bryant or Kevin Durant.","contentSnippet":"Luka Doncic spent his first 57 career games putting up numbers not seen since the days of a young LeBron James, Kobe Bryant or Kevin Durant.","guid":"26099269","isoDate":"2019-02-28T04:28:28.000Z"},{"title":"How former ref Tim Donaghy conspired to fix NBA games","link":"http://www.espn.com/nba/story/_/id/25980368/how-former-ref-tim-donaghy-conspired-fix-nba-games","pubDate":"Tue, 12 Feb 2019 12:00:54 EST","content":"In 2007, NBA ref Tim Donaghy pleaded guilty to betting on games he officiated. But it was never proved that he fixed them -- until now. Our two-year investigation reveals how he did it, whom he did it with and the millions that flowed from the conspiracy.","contentSnippet":"In 2007, NBA ref Tim Donaghy pleaded guilty to betting on games he officiated. But it was never proved that he fixed them -- until now. Our two-year investigation reveals how he did it, whom he did it with and the millions that flowed from the conspiracy.","guid":"25980368","isoDate":"2019-02-12T17:00:54.000Z"},{"title":"Predicting NBA superstar moves: KD, Kyrie, AD and Kawhi","link":"http://www.espn.com/nba/story/_/id/26024501/predicting-nba-superstar-moves-kd-kyrie-ad-kawhi","pubDate":"Fri, 22 Feb 2019 08:33:10 EST","content":"We polled the ESPN Forecast panel of experts for their votes on the next teams for star players.","contentSnippet":"We polled the ESPN Forecast panel of experts for their votes on the next teams for star players.","guid":"26024501","isoDate":"2019-02-22T13:33:10.000Z"}],"image":{"link":"http://www.espn.com","url":"https://a.espncdn.com/i/espn/teamlogos/lrg/trans/espn_dotcom_black.gif","title":"www.espn.com - NBA","width":"110","height":"110"},"title":"www.espn.com - NBA","description":"Latest NBA news from www.espn.com","generator":"ESPN Inc. http://www.espn.com","link":"http://www.espn.com","language":"en","copyright":"Copyright 2015 ESPN Inc.","lastBuildDate":"Sun, 3 Mar 2019 19:49:38 GMT","ttl":"30"}

import ChannelPage from '../pages/channel.js'

beforeAll(async () => {
  await preloadAll()
})

describe('Pages', () => {
  describe('Channel', () => {
    it('Should match with snapshot', () => {
      const component = renderer.create(<ChannelPage data={TEST_FEED} />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })  
})