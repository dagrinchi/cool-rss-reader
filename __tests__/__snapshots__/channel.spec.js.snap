// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Channel Should match with snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 OHJVR"
>
  <header
    className="AtAppBar__Container-um73xs-0 ibZoEd StyledBox-sc-13pk1d4-0 gXuzit"
    style={
      Object {
        "zIndex": "1",
      }
    }
  >
    <button
      className="AtButton__Button-sc-1xxhv8r-0 byhqJH StyledButton-sc-323bzc-0 lbUBVe"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <svg
        aria-label="LinkPrevious"
        className="StyledIcon-ofa7kd-0 jAaLrm"
        viewBox="0 0 24 24"
      >
        <path
          d="M2,12 L22,12 M13,3 L22,12 L13,21"
          fill="none"
          stroke="#000"
          strokeWidth="2"
          transform="matrix(-1 0 0 1 24 0)"
        />
      </svg>
    </button>
    <h2
      className="StyledHeading-sc-1rdh4aw-0 NPFk"
    >
      RSS Reader App
    </h2>
    <div
      style={
        Object {
          "width": "48px",
        }
      }
    />
  </header>
  <div
    className="StyledBox-sc-13pk1d4-0 IRSNj"
    style={
      Object {
        "marginTop": "70px",
      }
    }
  >
    <div
      className="channel__Container-sc-1xf8zdf-0 hPizmf StyledBox-sc-13pk1d4-0 IRSNj"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 IRSNj"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 IRSNj"
        >
          <img
            className="StyledImage-ey4zx9-0 knmSyL"
            src="https://a.espncdn.com/i/espn/teamlogos/lrg/trans/espn_dotcom_black.gif"
          />
        </div>
        <h1
          className="heading StyledHeading-sc-1rdh4aw-0 fXzHqj"
        >
          www.espn.com - NBA
        </h1>
      </div>
      <div
        className="OrFeedGrid__Container-vere28-0 kPmued StyledBox-sc-13pk1d4-0 hVqWZj"
      >
        <div
          className="StyledBox-sc-13pk1d4-0 kCrWVa"
        >
          Loading...
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 kCrWVa"
        >
          Loading...
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 kCrWVa"
        >
          Loading...
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 kCrWVa"
        >
          Loading...
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 kCrWVa"
        >
          Loading...
        </div>
        <div
          className="StyledBox-sc-13pk1d4-0 kCrWVa"
        >
          Loading...
        </div>
      </div>
    </div>
  </div>
</div>
`;
